pkg/provisioner/ironic/ironic.go

```diff
@@ -1784,6 +1790,14 @@ func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error
 	return result, nil
 }
 
+func (p *ironicProvisioner) ironicNodeNameFromHost() string {
+	if !p.watchAllNamespaces || p.host.Namespace == p.podNamespace {
```

Comment:
If we are watching all namespaces and a BMH resource is created in pod namespace, will this return just the hostname (without the namespace prefix)? 
Even in that scenario, if I understand this feature correctly, we would want this method to return <namespace>.<hostname> whenever p.watchAllNamespaces.

Reply:
so the point, as I understand it, is to prevent these names from clashing (because they are different clusters and you could have nodes with the same name). The idea is that if we go from watchAllNamespace to !watchAllNamespace and back at least the current cluster (it's bmh is in POD_NAMESPACE) would always be the same. I think @hardys suggested keeping the current namespaced names without the "." so that when we upgrade the names are consistent.

Reply:
When BMH resources are listed for all namespaces, the output would contain some with a namespace prefix and others without. Just making sure we are ok with it.

Reply:
In the thread below I wasn't suggesting we keep the current names, but that we should rename existing ironic nodes in the case where the configuration has gone from single-namespace to watchAllNamespaces.

E.g here https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L493..L498 we'd change the name whenever it doesn't match the expected format.

It'll need some testing to ensure there aren't corner cases to consider, but that would resolve this concern I think?

Obviously the code here would also need to return the prefixed name whenever the watchAllNamespaces mode is enabled.

Reply:
> The idea is that if we go from watchAllNamespace to !watchAllNamespace and back 

I was thinking that the only sane path was to switch from a single-namespace to `watchAllNamespaces=true` - that was the focus of my previous comment, I don't see how we can really support the reverse path, since there could be BMH resources in any namespace?

Reply:
Thanks for the clarifcation @hardys , I'll fixup today.